client/checkout/api/index.js

```diff
@@ -231,21 +238,24 @@ export default class WCPayAPI {
 				throw response.data.error;
 			}
 
-			// eslint-disable-next-line camelcase
-			const { client_secret } = response.data;
-			return this.getStripe()
-				.confirmCardSetup( client_secret, {
+			if ( response.data.status === 'succeeded' ) {
+				// No need for further authentication.
+				return response.data;
+			}
```

Comment:
Oh, I hadn't seen that backstory. It looks like the conclusion was "let's always rely on `confirmCardSetup` because it will make code simpler", but it wasn't the end of the world if it was implemented the other way.

I haven't experienced that the new code is much more complex, and it saves a round-trip on cards that don't require 3DS (which are the majority right now), so I stand by my decision :)

Reply:
I agree, it definitely makes sense to do it this way. I just added that comment to give some context 😄